File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        students = deque(students)
        sandwiches = deque(sandwiches)
        while students:
            if students[0] == sandwiches[0]:
                students.popleft()
                sandwiches.popleft()
            elif students.count(sandwiches[0]) == 0:
                break   
            else:
                students.append(students.popleft())                
         
        return len(students)

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.go
func countStudents(students []int, sandwiches []int) int {
    passed := 0
    for len(students) != 0 && passed != len(students) {
        if students[0] != sandwiches[0] {
            passed++
            students = append(students[1:], students[0])
        } else {
            passed = 0
            students = students[1:]
            sandwiches = sandwiches[1:]
        }
    }
    return passed
}